text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,369 @@
+|🌐 | `Ember.ObjectProxy` | `import ObjectProxy from '@ember/object/proxy';` |
+|🧷 | `Ember.HistoryLocation` | `import HistoryLocation from '@ember/routing/history-location';` |
+|🧷 | `Ember.HashLocation` | `import HashLocation from '@ember/routing/hash-location';` |
+|🧷 | `Ember.NoneLocation` | `import NoneLocation from '@ember/routing/none-location';` |
+|🌐 | `Ember.Route` | `import Route from '@ember/routing/route';` |
+|🌐 | `Ember.run` | `import { run } from '@ember/runloop';` |
+|🌐 | `Ember.Service` | `import Service from '@ember/service';` |
+|🌐 | `Ember.compare` | `import { compare } from '@ember/utils';` |
+|🌐 | `Ember.isBlank` | `import { isBlank } from '@ember/utils';` |
+|🌐 | `Ember.isEmpty` | `import { isEmpty } from '@ember/utils';` |
+|🌐 | `Ember.isEqual` | `import { isEqual } from '@ember/utils';` |
+|🌐 | `Ember.isPresent` | `import { isPresent } from '@ember/utils';` |
+|🌐 | `Ember.typeOf` | `import { typeOf } from '@ember/utils';` |
+|🌐 | `Ember._getComponentTemplate` | `import { getComponentTemplate } from '@ember/component';` | 
+|🌐 | `Ember._setComponentTemplate` | `import { setComponentTemplate } from '@ember/component';` | 
+|🌐 | `Ember._helperManagerCapabilities` | `import { capabilities } from '@ember/helper';` | 
+|🌐 | `Ember._setHelperManager` | `import { setHelperManager } from '@ember/helper';` | 
+|🌐 | `Ember._setModifierManager` | `import { setModifierManager } from '@ember/modifier';` | 
+|🌐 | `Ember._templateOnlyComponent` | `import templateOnly from '@ember/component/template-only';` | 
+|🌐 | `Ember._invokeHelper` | `import { invokeHelper } from '@ember/helper';` | 
+|🌐 | `Ember._hash` | `import { hash } from '@ember/helper';` | 
+|🌐 | `Ember._array` | `import { array } from '@ember/helper';` | 
+|🌐 | `Ember._concat` | `import { concat } from '@ember/helper';` | 
+|🌐 | `Ember._get` | `import { get } from '@ember/helper';` | 
+|🌐 | `Ember._on` | `import { on } from '@ember/modifier';` | 
+|🌐 | `Ember._fn` | `import { fn } from '@ember/helper';` | 
+|🌐 | `Ember.ENV` | `import MyEnv from '<my-app>/config/environment';` |
+
+
+[RFC-615]: https://rfcs.emberjs.com/id/0615-autotracking-memoization
+
+## How We Teach This
+
+The guides already use the modern imports where available.
+
+There is a place that needs updating, around advanced debugging, where folks configure Backburner to be in debug mode.
+- https://guides.emberjs.com/release/applications/run-loop/#toc_where-can-i-find-more-information
+- https://guides.emberjs.com/release/configuring-ember/debugging/#toc_errors-within-emberrunlater-backburner
+  - Access to backburner here isn't relevant though because it's accessed from the `run` import from `@ember/runloop`
+
+When using embroider and `staticEmberSource: true`, the benefits of not having this file can be realized in apps (as long as the app and all consumed addons do not import from 'ember')
+
+Available Codemods
```

Comment:
Can we have a babel transform? This would be a severe pain to update any orphaned packages

Reply:
I started adding ember-observer usages to the table where the behavior has a direct swappable replacement and... I'm _shocked_ how many addons are still using stuff this old.

Some sort of build-time transform specifically for old addons using `Ember.` seems appropriate -- maybe as an opt in, as to not slow down new projects

Reply:
Even if we just had some unofficially documented fallback using something https://www.npmjs.com/package/babel-plugin-transform-imports would probably be fine